mooncake-transfer-engine/src/transport/coro_rpc_connector/cororpc_communicator.cpp

```diff
@@ -0,0 +1,272 @@
+}
+
+bool CoroRPCCommunicator::initialize(const Config& config) {
+    impl_->config = config;
+
+    if (!config.listen_address.empty()) {
+        std::cout << "Initializing server on " << config.listen_address
+                  << std::endl;
+
+        impl_->server_ = std::make_unique<coro_rpc::coro_rpc_server>(
+            config.thread_count, config.listen_address,
+            std::chrono::seconds(config.timeout_seconds));
+
+        impl_->server_->register_handler<
+            &CoroRPCCommunicator::Impl::handleDataTransfer,
+            &CoroRPCCommunicator::Impl::handleTensorTransfer>(impl_.get());
+    }
+
+    std::cout << "Communicator initialized with client pool support"
+              << std::endl;
+    return true;
+}
+
+bool CoroRPCCommunicator::startServer() {
+    if (!impl_->server_ || impl_->config.listen_address.empty()) return false;
+
+    try {
+        auto ec = impl_->server_->start();
+        if (ec.val() == 0) {
+            impl_->is_server_started = true;
+            std::cout << "Server started on " << impl_->config.listen_address
+                      << std::endl;
+            return true;
+        } else {
+            std::cerr << "Failed to start server: " << ec.message()
+                      << std::endl;
+            return false;
+        }
+    } catch (const std::exception& e) {
+        std::cerr << "Failed to start server: " << e.what() << std::endl;
+        return false;
+    }
+}
+
+bool CoroRPCCommunicator::startServerAsync() {
+    if (!impl_->server_ || impl_->config.listen_address.empty()) return false;
+
+    try {
+        auto ec = impl_->server_->async_start();
+        if (!ec.hasResult()) {
+            impl_->is_server_started = true;
+            std::cout << "Server started asynchronously on "
+                      << impl_->config.listen_address << std::endl;
+            return true;
+        } else {
+            std::cerr << "Failed to start server asynchronously" << std::endl;
+            return false;
+        }
+    } catch (const std::exception& e) {
+        std::cerr << "Failed to start server asynchronously: " << e.what()
+                  << std::endl;
+        return false;
+    }
+}
+
+void CoroRPCCommunicator::stopServer() {
+    if (impl_->is_server_started) {
+        impl_->is_server_started = false;
+        std::cout << "Server stopped" << std::endl;
+    }
+}
+
+int CoroRPCCommunicator::sendData(const std::string& target_address,
+                                  const void* data, size_t data_size) {
+    auto result = async_simple::coro::syncAwait(
+        sendDataAsync(target_address, data, data_size));
+    return result.code;
+}
+
+async_simple::coro::Lazy<result> CoroRPCCommunicator::sendDataAsync(
+    const std::string& target_address, const void* data, size_t data_size) {
+    try {
+        std::string_view data_view(static_cast<const char*>(data), data_size);
+
+        auto rpc_result = co_await client_pools_.send_request(
```

Comment:
deal with rpc_result:
if (!rpc_result.has_value()) {{
   // not conneted, err msg: ec.error
}